lxd/patches.go

```diff
@@ -2247,6 +2248,63 @@ func patchStorageApiInsertZfsDriver(name string, d *Daemon) error {
 	return nil
 }
 
+func patchStorageZFSnoauto(name string, d *Daemon) error {
+	pools, err := dbStoragePools(d.db)
+	if err != nil {
+		if err == NoSuchObjectError {
+			return nil
+		}
+		logger.Errorf("Failed to query database: %s", err)
+		return err
+	}
+
+	for _, poolName := range pools {
+		_, pool, err := dbStoragePoolGet(d.db, poolName)
+		if err != nil {
+			logger.Errorf("Failed to query database: %s", err)
+			return err
+		}
+
+		if pool.Driver != "zfs" {
+			continue
+		}
+
+		zpool := pool.Config["zfs.pool_name"]
+		if zpool == "" {
+			continue
+		}
+		path := fmt.Sprintf("%s/containers", zpool)
+
+		output, err := shared.RunCommand(
```

Comment:
I first tried that approach but it ended up looking more complicated than just poking zfs for everything under containers/. That's mostly because with this approach I don't have to care about parsing container names to skip snapshots.